### PR TITLE
[WFLY-19359] define kafka yaml file for OpenShift and update the README.

### PR DESCRIPTION
**Pull request by kstekovi**
https://issues.redhat.com/browse/WFLY-19408

Define AMQ stream operator, instance and topic in a  yaml file for OpenShift and update the README.

### CODE REVIEW

**Comment by emmartins** [comment]
@kstekovi thanks, @kabir please review

**Review by kabir — CHANGES_REQUESTED** [review]
The CI should also be updated to use this file.

OpenShift: 
https://github.com/wildfly/quickstart/blob/main/.ci/openshift-ci/build-root/scripts/qs-overrides/microprofile-reactive-messaging-kafka/overridable-functions.sh#L66-L106

Kubernetes: 
https://github.com/wildfly/quickstart/blob/main/.github/workflows/scripts/kubernetes/qs-overrides/microprofile-reactive-messaging-kafka/overridable-functions.sh#L56-L95

I see in the CI code for both there is a 'pause' between the cluster and creating the topic. But I don't know if that is actually needed or not.

For OpenShift, I am pretty sure the break between creating the subscription and the cluster is needed. Unless it handles things differently for some reason when using one file.

The Kubernetes part is not in the README yet, it is part of https://github.com/wildfly/quickstart/pull/919 (@emmartins )

**Comment by kstekovi** [comment]
Thanks for review @kabir 

Could you please take a look again? The CI is updated.

**Review by kabir — CHANGES_REQUESTED** [review]
Hi,

The changes **look** good but it doesn't run :-)

If you see the original OpenShift setup test script it used to install the operator, but then it needs to wait for the AMQ Streams operator to start and make the CRDs available. That was done here https://github.com/wildfly/quickstart/pull/929/files#diff-3002e707de1a81994b7bfc2d03e271afd1e186c1d8ee4c8ae7c70992bbddd5aaL62-L65. Later there is a check of the exit value from the oc calls to create the cluster. Once the oc call to create the cluster exits with 0 we can proceed.

As it is now, if I try to apply the file it fails when trying to install the cluster:

```
% oc apply -f ./charts/kafka-on-openshift.yaml --wait --timeout=10m0s
subscription.operators.coreos.com/amq-streams unchanged
resource mapping not found for name: "my-cluster" namespace: "" from "./charts/kafka-on-openshift.yaml": no matches for kind "Kafka" in version "kafka.strimzi.io/v1beta2"
ensure CRDs are installed first
resource mapping not found for name: "testing" namespace: "" from "./charts/kafka-on-openshift.yaml": no matches for kind "KafkaTopic" in version "kafka.strimzi.io/v1beta2"
ensure CRDs are installed first
```

So I think this needs to be two scripts. One for the subscription, and one for the cluster/topic. And there should b

**Comment by kstekovi** [comment]
Hi, This didn't happen on my cluster. I try also different one and hit this problem. 

So i split resource definition into two yaml files which are processed separately as you suggest. Thanks.

**Review by kabir — CHANGES_REQUESTED** [review]
Hi @kstekovi A few small changes needed, but apart from that I've verified that both the README instructions and running the tests work against a Clusterbot instance.

So I think once you make the fixes I suggested, this can be merged

File: microprofile-reactive-messaging-kafka/README-source.adoc [hunk 1]
```diff
@@ -987,6 +987,9 @@ bin/kafka-topics.sh --create --topic testing --bootstrap-server localhost:9092
 
 // OpenShift
 include::../shared-doc/build-and-run-the-quickstart-with-openshift.adoc[leveloffset=+1]
+----
+$ oc delete -f ./charts/kafka-on-openshift.yaml --wait --timeout=10m0s
```
[the remark this answers is not in the document]
Comment (kstekovi) [reply microprofile-reactive-messaging-kafka/README-source.adoc:991]: I miss this. Instruction updated.

File: microprofile-reactive-messaging-kafka/helm-install-prerequisites.adoc [hunk 2]
```diff
@@ -1,67 +1,46 @@
 === Install AMQ Streams on OpenShift
 
 The functionality of this quickstart depends on a running instance of the
-https://access.redhat.com/products/red-hat-amq#streams[AMQ Streams] Operator. AMQ Streams is a Red Hat project based on Apache Kafka. To deploy AMQ Streams in the Openshift environment:
+https://access.redhat.com/products/red-hat-amq#streams[AMQ Streams] Operator. AMQ Streams is a Red Hat project based
+on Apache Kafka. To deploy AMQ Streams in the Openshift environment:
```
[the remark this answers is not in the document]
Comment (kstekovi) [reply microprofile-reactive-messaging-kafka/helm-install-prerequisites.adoc]: line break removed.

File: microprofile-reactive-messaging-kafka/helm-install-prerequisites.adoc [hunk 3]
```diff
@@ -1,67 +1,46 @@
...
-spec:
-  partitions: 3
-  replicas: 3
-EOF
+$ oc apply -f ./charts/kafka-on-openshift.yaml --wait --timeout=10m0s
 ----
 
-Although the above commands will return pretty immediately, your AMQ Streams instance will not be available until its entity operator is up and running. The name of the pod will be of the format `my-cluster-entity-operator-xxxxxxxxx-yyyyy`.
+Although the above commands will return pretty immediately, your AMQ Streams instance will not be available until its
+entity operator is up and running. The name of the pod will be of the format `my-cluster-entity-operator-xxxxxxxxx-yyyyy`.
```
[the remark this answers is not in the document]
Comment (kstekovi) [reply microprofile-reactive-messaging-kafka/helm-install-prerequisites.adoc]: line break removed.

**Review by kabir — APPROVED** [review]

**Pull request merged**